api/data_explorer/controllers/export_url_controller.py

```diff
@@ -165,32 +165,58 @@ def _get_range_clause(column, value):
     return column + " >= " + str(low) + " AND " + column + " < " + str(high)
 
 
-def _get_clause(column, type, value):
+def _get_clause(column, field_type, value, is_sample_file_column):
     """Returns a single condition of a WHERE clause,
     eg "((age76 >= 20 AND age76 < 30) OR (age76 >= 30 AND age76 < 40))".
     """
-    if type == 'text':
-        clause = column + ' = "' + value + '"'
-    elif type == 'boolean':
-        clause = column + ' =' + value
+    if is_sample_file_column:
+        clause = '%s IS NOT NULL' % column
+    elif field_type == 'text':
+        clause = '%s = "%s"' % (column, value)
+    elif field_type == 'boolean':
+        clause = '%s = %s' % (column, value)
     else:
         clause = _get_range_clause(column, value)
     return clause
 
 
+def _get_facet_and_value(filter):
+    facets = current_app.config['UI_FACETS']
+    split = filter.rsplit('=', 1)
+    return facets[split[0]], split[1]
+
+
+def _get_table_name_column(es_field_name, sample_file_column_fields):
+    is_sample_file_column = False
+    if es_field_name.startswith('samples.'):
+        es_field_name = es_field_name.replace('samples.', '')
+        # Check if this is one of the special '_has_<file_type>' facets.
+        stripped = es_field_name.replace('_has_', '')
+        if stripped in sample_file_column_fields:
+            es_field_name = sample_file_column_fields[stripped]
+            is_sample_file_column = True
```

Comment:
`is_sample_file_column` sounds like "is one of the columns in bigquery.json sample_file_columns, eg bryancrampton-testing.data_explorer.1000_genomes_sample_info.chr1_vcf

How about `is_has_sample_file_type_field`

Reply:
Done (`has_sample_file_type_field`)